packages/SwingSet/test/upgrade/test-upgrade.js

```diff
@@ -179,7 +179,10 @@ const testUpgrade = async (
   t.deepEqual(v2result.youAre, 'v2');
   t.deepEqual(krefOf(v2result.marker), krefOf(marker));
   t.deepEqual(v2result.data, ['some', 'data']);
-  t.deepEqual(v2result.upgradeResult, { incarnationNumber: 2 });
+  t.deepEqual(v2result.upgradeResult, {
+    incarnationNumber: 2,
+    rootObject: v2result.upgradeResult?.rootObject,
```

Comment:
If we're going to re-supply the root object, we should assert that it's the same identity as the original, else this test is also contributing to the confusion.

`kunser` retains a module-level Map of the ersatz Presences it creates, so we just need the original `buildV1` to return us `ulrikRoot` along with everything else, then in `testUpgrade` we can stash it. Then this comparison changes `rootObject: v2result.upgradeResult?.rootObject` into `rootObject: ulrikRoot`.

Reply:
Hrm, when I check the analogous situation in the new null-upgrade test I see that root object identity is not preserved:
```
  Difference (- actual, + expected):

    Object @Alleged: undefined {
  -   getKref: Function getKref {},
  +   getKref: Function getKref {},
  -   iface: Function iface {},
  +   iface: Function iface {},
    }
```
But all the other assertions still pass and the result from `getKref()` _does_ match, so I'm going to revert the `processUpgradeVat` changes.